lib/emailbutler/helpers.rb

```diff
@@ -23,6 +23,7 @@ def serialize(value, reverse=true)
 
     def serialize_value(value)
       return value.to_global_id.to_s if value.respond_to?(:to_global_id)
+      return nil unless value&.valid_encoding?
```

Comment:
@zachfeldman thank you for your PR, but I have a few questions/notes

- you can write `return unless value&.valid_encoding?`
- if value is any kind of number - this line fails
- can you provide some anonymous example of params of your mailer?

this method is required for saving information about sent emails, so if skip saving some params -> then you probably could not resend it
but I think, maybe I'll add configuration for disabling resending feature, and use only logging mailer params (with skipping some of them)

Reply:
@kortirso thanks for the fast response!

- Yes, sorry I'm used to my liner picking up things like this
- I've re-scoped it to just `String`s
- Sure, the `value` that I fail on is basically any PDF encoded as an attachment to the email. The string looks something like this (truncated): `%PDF-1.4\n%\xE2\xE3\xCF\xD3\n3 0 obj\n[/PDF/Text/ImageB/ImageC/ImageI]\nendobj\n4 0 obj\n<</Type/Font/Subtype/Type1/BaseFont/Helvetica/Encoding/MacRomanEncoding>>\nendobj\n5 0 obj\n<`
- If you have suggestions on where I could/should add a spec to test this, I'm glad to do so!

Reply:
you can add test to `spec/mailers/test_mailer_spec.rb`, to mailer_params or action_params on line 7, and check saved result on line 20 (that such param is not present in saved params)

I think right now it is good solution for fixing this issue

while you checked class of value then you can do `return if value.is_a?(String) && !value.valid_encoding?`

Reply:
@kortirso thanks, I fixed the syntax as suggested. I'm trying to add the spec but not able to load the test database. There doesn't appear to be a schema or migrations in the dummy app. I created a DB but I can't seem to load anything in it. Any pointers?
```
1) TestMailer#send_simple_email creates Message object
     Failure/Error: let!(:message) { create :emailbutler_message }
     
     ActiveRecord::StatementInvalid:
       PG::UndefinedTable: ERROR:  relation "emailbutler_messages" does not exist
       LINE 10:  WHERE a.attrelid = '"emailbutler_messages"'::regclass
                                    ^
     # ./spec/mailers/test_mailer_spec.rb:6:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:38:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:37:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # PG::UndefinedTable:
     #   ERROR:  relation "emailbutler_messages" does not exist
     #   LINE 10:  WHERE a.attrelid = '"emailbutler_messages"'::regclass
     #                                ^
     #   ./spec/mailers/test_mailer_spec.rb:6:in `block (3 levels) in <top (required)>'

```

Reply:
rails db:drop
rails db:create
rails db:schema:load

while there is dummy app in tests you can run rake tasks

Reply:
@zachfeldman I'll merge your PR, tahnk you for this issue and integrating emailbutler, I'll add test

Reply:
Sounds good @kortirso , thank you!